src/lib/util.js

```diff
@@ -1,11 +1,50 @@
 export function isValidRepoUrl(url) {
-  const regex = new RegExp("https?://github.com/", "i"); // passing "i" as 2nd argument to ignore case
-  const isFullUrl = regex.test(url);
-  const relativeRepoUrl = isFullUrl ? url.toLowerCase().replace(regex, "") : url.toLowerCase();
+  url = url.trim().toLowerCase();
+  url = url.substr(0, 1) === "/" ? url.substr(1) : url;
 
-  const [owner, repo] = relativeRepoUrl.split("/");
-  if (!owner || !repo) {
+  const isRelativeUrl = !(url.substr(0, 4) === "http" || url.includes(".com") || url.includes("www."));
+  if (isRelativeUrl) {
+    return relativeUrlValidator(url);
+  }
+
+  return absoluteUrlValidator(url);
+
+}
+
+function relativeUrlValidator(url) {
+  try {
+    const githubLink = "github.com/";
+    const newUrl = "https://" + githubLink + url;
+    const urlObject = new URL(newUrl);
+
+    const [owner, repo] = url.split("/");
+    if (!owner || !repo || !(urlObject.protocol === "http:" || urlObject.protocol === "https:")) {
+      return [false, null];
+    }
+    return [true, url];
+  } catch (failedToConstructURL) {
+    return [false, null];
+  }
+}
+
+function absoluteUrlValidator(url) {
+  try {
+    const githubLink = "github.com/";
+    const urlObject = new URL(url);
+
+    const getIndexGithub = url.indexOf(githubLink) + githubLink.length;
+    const relativeRepoUrl = url.substr(getIndexGithub, url.length);
+
+    const [owner, repo] = relativeRepoUrl.split("/");
+
+    if (!url.includes("github.com"))
+      return [false, null];
+
+    if (!owner || !repo || !(urlObject.protocol === "http:" || urlObject.protocol === "https:")) {
+      return [false, null];
+    }
+    return [true, owner + "/" + repo];
+  } catch (failedToConstructURL) {
```

Comment:
Good to see, I saw the `_` earleir and was going to ask about that.

Reply:
I almost think it should be named newUrlThrewError, but failedToConstructURL seems fine as-is. Yeah I agree - the _ seemed like bad practice.